src/main/scala/centaur/test/Test.scala

```diff
@@ -99,23 +99,29 @@ object Operations {
     }
   }
 
-  def retrieveMetadata(workflow: SubmittedWorkflow): Test[WorkflowMetadata] = {
-    new Test[WorkflowMetadata] {
-      override def run: Try[WorkflowMetadata] = {
-        val response = MetadataRequest(Get(CentaurConfig.cromwellUrl + "/api/workflows/v2/" + workflow.id + "/metadata"))
-        // Try to convert the response to a Metadata in our return Try.
-        // Currently any error msg will be opaque as it's unlikely to be an issue (can flesh out later)
-        sendReceiveFutureCompletion(response map { r => WorkflowMetadata.fromMetadataJson(r).toOption.get })
-      }
-    }
-  }
 
-  def validateMetadata(retrievedMetadata: WorkflowMetadata, expectedMetadata: WorkflowMetadata, workflowID: UUID): Test[Unit] = {
+
+
+  def validateMetadata(workflow: SubmittedWorkflow, expectedMetadata: WorkflowMetadata): Test[Unit] = {
+    val consistencyTimeout = Future { Thread.sleep(CentaurConfig.metadataConsistencyTimeout.toMillis) }
+
     new Test[Unit] {
+      def validateMetadataUntilTimeout(workflow: SubmittedWorkflow, expectedMetadata: WorkflowMetadata): Try[Unit] = {
+          val response = MetadataRequest(Get(CentaurConfig.cromwellUrl + "/api/workflows/v2/" + workflow.id + "/metadata"))
+          // Try to convert the response to a Metadata in our return Try.
+          // Currently any error msg will be opaque as it's unlikely to be an issue (can flesh out later)
+          val metadata = sendReceiveFutureCompletion(response map { r => WorkflowMetadata.fromMetadataJson(r).toOption.get })
+          metadata.map(expectedMetadata.diff(_, workflow.id)) match {
+            case Success(d) if d.isEmpty => Success()
+            case Success(d) if consistencyTimeout.isCompleted =>
+              Failure(throw new Exception(s"Invalid metadata response:\n -${d.mkString("\n -")}\n"))
+            case Success(_) =>  validateMetadataUntilTimeout(workflow, expectedMetadata)
+            case Failure(e) => Failure(e)
```

Comment:
if you're just passing the failure through, maybe the above makes more sense as a `map`